homeassistant/components/usb/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""The USB Discovery integration."""
+from __future__ import annotations
+
+import dataclasses
+import datetime
+import logging
+import sys
+
+from serial.tools.list_ports import comports
+from serial.tools.list_ports_common import ListPortInfo
+
+from homeassistant import config_entries
+from homeassistant.const import EVENT_HOMEASSISTANT_STARTED, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.loader import async_get_usb
+
+from .flow import FlowDispatcher, USBFlow
+from .models import USBDevice
+from .utils import usb_device_from_port
+
+_LOGGER = logging.getLogger(__name__)
+
+# Perodic scanning only happens on non-linux systems
+SCAN_INTERVAL = datetime.timedelta(minutes=60)
```

Comment:
Does it make sense to do this hourly? Maybe we can just have the frontend trigger a scan via a WS when the integration tab is opened ? That way we could also include Plex discovery.

Reply:
I figure there are cases where the user doesn't know what to do, plugs in the stick and magically an hour later it appears.

Reply:
Also we only scan on windows and Mac.  On Linux we can observe the device appearing.

Reply:
Switch to websocket here: https://github.com/home-assistant/core/pull/54953

Need to figure out where to tie in the frontend